src/table.rs

```diff
@@ -92,15 +92,15 @@ impl Table {
     }
 
     /// Fetch current table metadata.
-    pub fn current_table_metadata(&self) -> Result<&types::TableMetadata> {
+    pub fn current_table_metadata(&self) -> &types::TableMetadata {
         assert!(
             self.current_version != 0,
             "table current version must be valid"
         );
 
         self.table_metadata
             .get(&self.current_version)
-            .ok_or_else(|| anyhow!("table metadata not found"))
+            .expect("table metadata of current version must be exist")
```

Comment:
Correct me if I'm wrong: There are two case to update table_meadata:
1. load when open 
2. commit write
We gurantee that if above operation is successful, the table_metadata should have the metadata and according version.

Based on this, we can use expect here.

Reply:
Both LGTM, let's keep moving and listen to users' feedback later.